subprojects/core-api/src/main/java/org/gradle/api/file/FileCollection.java

```diff
@@ -145,7 +145,10 @@
      *
      * @return this
      * @throws StopExecutionException When this collection is empty.
+     *
+     * @deprecated Throw a `StopExecutionException` in your code manually instead.
```

Comment:
Should we mention `@SkipWhenEmpty` here as well? It might work in many cases and is even better than using exceptions to control execution.

Reply:
Good point. I updated the release notes and the Javadoc.